### PR TITLE
feat: Implement address format

### PR DESCRIPTION
**Pull request by duanyytop**

### CODE REVIEW

File: utils/src/main/java/org/nervos/ckb/address/AddressUtils.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import org.nervos.ckb.utils.Bech32;
+import org.nervos.ckb.utils.Network;
+import org.nervos.ckb.utils.Numeric;
+
+/**
+ * Created by duanyytop on 2019-04-18.
+ * Copyright © 2018 Nervos Foundation. All rights reserved.
+ *
+ * AddressUtils based on CKB AddressUtils Format [RFC](https://github.com/nervosnetwork/rfcs/blob/4f87099a0b1a02a8bc077fc7bea15ce3d9def120/rfcs/0000-address-format/0000-address-format.md),
+ * and [Common AddressUtils Format](https://github.com/nervosnetwork/ckb/wiki/Common-Address-Format).
```
Comment (ashchan) [inline utils/src/main/java/org/nervos/ckb/address/AddressUtils.java]:
```suggestion
 * and [Common Address Format](https://github.com/nervosnetwork/ckb/wiki/Common-Address-Format).
```

File: utils/src/main/java/org/nervos/ckb/address/AddressUtils.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import org.nervos.ckb.exceptions.AddressFormatException;
+import org.nervos.ckb.utils.Bech32;
+import org.nervos.ckb.utils.Network;
+import org.nervos.ckb.utils.Numeric;
+
+/**
+ * Created by duanyytop on 2019-04-18.
+ * Copyright © 2018 Nervos Foundation. All rights reserved.
+ *
+ * AddressUtils based on CKB AddressUtils Format [RFC](https://github.com/nervosnetwork/rfcs/blob/4f87099a0b1a02a8bc077fc7bea15ce3d9def120/rfcs/0000-address-format/0000-address-format.md),
```
Comment (ashchan) [inline utils/src/main/java/org/nervos/ckb/address/AddressUtils.java]:
```suggestion
 * AddressUtils based on CKB Address Format [RFC](https://github.com/nervosnetwork/rfcs/blob/4f87099a0b1a02a8bc077fc7bea15ce3d9def120/rfcs/0000-address-format/0000-address-format.md),
```

File: utils/src/test/java/org/nervos/ckb/crypto/Bech32Test.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                "tb1qrp33g0q5c5txsp9arysrx4k6zdkfs4nce4xj0gdcccefvpysxf3q0sl5k7",
+                "bc1pw508d6qejxtdg4y5r3zarvary0c5xw7kw508d6qejxtdg4y5r3zarvary0c5xw7k7grplx",
+                "BC1SW50QA3JX3S",
+                "bc1zw508d6qejxtdg4y5r3zarvaryvg6kdaj",
+                "tb1qqqqqp399et2xygdj5xreqhjjvcmzhxw4aywxecjdzew6hylgvsesrxh6hy"
+        };
+        for (String address: validAddresses) {
+            try {
+                Bech32.decode(address);
+            } catch (AddressFormatException e) {
```
Comment (classicalliu) [inline utils/src/test/java/org/nervos/ckb/crypto/Bech32Test.java]: Remove `catch` here

File: utils/src/main/java/org/nervos/ckb/address/AddressUtils.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    public String strToAscii(String value) {
+        StringBuilder sb = new StringBuilder();
+        char[] chars = value.toCharArray();
+        for (int i = 0; i < chars.length; i++) {
+            sb.append(Integer.toHexString((int)chars[i]));
+        }
+        return sb.toString();
+    }
+
+    public byte[] convertBits(byte[] data, int fromBits, int toBits, boolean pad) {
```
[the remark this answers is not in the document]
Comment (duanyytop) [reply utils/src/main/java/org/nervos/ckb/address/AddressUtils.java]: Reference Swift and JavaScript implement code.

File: utils/src/main/java/org/nervos/ckb/address/AddressUtils.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+            byte value = data[i];
+            accumulator = (((accumulator & 0xff) << fromBits) | (value & 0xff));
+            bits += fromBits;
+            while (bits >= toBits) {
+                bits -= toBits;
+                result[index] = (byte) ((accumulator >> bits) & mask);
+                ++index;
+            }
+        }
+        if (!pad) {
```
[the remark this answers is not in the document]
Comment (duanyytop) [reply utils/src/main/java/org/nervos/ckb/address/AddressUtils.java]: I see. But if `!pad` is main flow and `pad` is error flow, I think this flow will more clear.

File: utils/src/test/java/org/nervos/ckb/crypto/Bech32Test.java (hunk 3 above)
Comment (duanyytop) [reply utils/src/test/java/org/nervos/ckb/crypto/Bech32Test.java]: I will refactor it.

**Pull request merged**